packages/common/src/createContract.ts

```diff
@@ -47,6 +53,7 @@ export type CreateContractOptions<
   TPublicClient extends PublicClient<TTransport, TChain>,
   TWalletClient extends WalletClient<TTransport, TChain, TAccount>
 > = Required<GetContractParameters<TTransport, TChain, TAccount, TAbi, TPublicClient, TWalletClient, TAddress>> & {
+  getResourceSelector: (functionSelector: Hex) => Promise<Hex>;
```

Comment:
since we already have a conditional below to figure out if we should use callFrom, we could make this arg optional to avoid this becoming a breaking change